From 2886452ad8d391d17580c1c6906fa33290a3d32b Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 11 Mar 2015 19:07:42 +0800
Subject: [PATCH 0214/5242] MLK-10406 ARM: imx: check the clk_set_rate()
 return value

commit  4428be8be603809b9ac209fbb4184204aec310c8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add return value check for clk_set_rate() to avoid set clock
rate fail.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: bcd9d743ef534cf58d2d007e6333270be12c9f06)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index ffa4482..e313de3 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -2257,8 +2257,13 @@ static int imx_uart_probe(struct platform_device *pdev)
 	}
 
 	sport->port.uartclk = clk_get_rate(sport->clk_per);
-	if (sport->port.uartclk > IMX_MODULE_MAX_CLK_RATE)
-		clk_set_rate(sport->clk_per, IMX_MODULE_MAX_CLK_RATE);
+	if (sport->port.uartclk > IMX_MODULE_MAX_CLK_RATE) {
+		ret = clk_set_rate(sport->clk_per, IMX_MODULE_MAX_CLK_RATE);
+		if (ret < 0) {
+			dev_err(&pdev->dev, "clk_set_rate() failed\n");
+			return ret;
+		}
+	}
 	sport->port.uartclk = clk_get_rate(sport->clk_per);
 
 	/* For register access, we only need to enable the ipg clock. */
-- 
1.7.9.5

